openraft/tests/life_cycle/t20_initialization.rs

```diff
@@ -229,6 +230,44 @@ async fn initialize_err_not_allowed() -> anyhow::Result<()> {
     Ok(())
 }
 
+/// What does this test do?
+///   this test is ought to test if the router is aware of network failures
+///
+/// - initialize 1 node as leader and 2 nodes as follower
+/// - set the router to be network failed
+/// - let the router connect to other nodes, assert errors
+#[async_entry::test(worker_threads = 8, init = "init_default_ut_tracing()", tracing_span = "debug")]
+async fn router_network_failure_aware() -> anyhow::Result<()> {
+    let config = Arc::new(Config::default().validate()?);
+    let mut router = RaftRouter::new(config.clone());
+    let nodes = btreeset! {0, 1, 2};
+    router.new_nodes_from_single(nodes.clone(), btreeset! {}).await?;
+
+    tracing::info!("--- fail router's network");
+    {
+        router.set_network_failure();
+    }
+
+    tracing::info!("--- assert if router is aware of network failure");
+    {
+        for node in nodes.into_iter() {
+            let resp = router.connect(node, None).await;
+            match resp {
+                Ok(_) => assert!(resp.is_err()), // make test fail
+                Err(err) => {
+                    let want = openraft::error::NetworkError::new(&anyerror::AnyError::error(format!(
+                        "failed to connect: {}",
+                        node
+                    )));
+                    assert_eq!(err, want);
+                }
+            };
+        }
+    }
+
+    Ok(())
+}
```

Comment:
This test only tests the error will be emitted. A test has to examine if a raft cluster works as expected. 

Hence `router.network_failed` has to be a btreemap to specify to which node it will fail to connect.